### PR TITLE
add concurrent search test for hnsw & hgraph

### PR DESCRIPTION
**Pull request by LHT129**
- introduce fixtures::ThreadPool
- test hgraph by enable SUPPORT_SEARCH_CONCURRENT feature

### CODE REVIEW

**Review by inabao — APPROVED** [review]
LGTM

File: tests/test_index.cpp [hunk 1]
```diff
@@ -253,5 +254,52 @@ TestIndex::TestSerializeFile(const IndexPtr& index_from,
         }
     }
 }
+void
+TestIndex::TestConcurrentKnnSearch(const TestIndex::IndexPtr& index,
+                                   const TestDatasetPtr& dataset,
+                                   const std::string& search_param,
+                                   float recall,
```
Comment (wxyucs) [inline tests/test_index.cpp:261]: rename recall -> expected_recall

**Review by wxyucs — CHANGES_REQUESTED** [review]

File: tests/test_index.cpp (hunk 1 above)
Comment (LHT129) [reply tests/test_index.cpp:261]: rename in another pr for all test recall

**Review by wxyucs — APPROVED** [review]
lgtm

**Pull request merged**